source/jormungandr/jormungandr/olympic_site_params_manager.py

```diff
@@ -111,32 +135,45 @@ def get_json_content(self, s3_object):
             logger.exception('Error while loading file: {}'.format(s3_object.key))
             return {}
 
-    def fill_olympic_site_params_from_s3(self):
+    @cache.memoize(app.config[str('CACHE_CONFIGURATION')].get(str('FETCH_S3_DATA_TIMEOUT'), 24 * 60))
+    def load_data(self, resource_s3_object):
+        json_content = self.get_json_content(resource_s3_object.s3_object)
+        self.str_datetime_time_stamp(json_content)
+        return json_content
+
+    @memory_cache.memoize(
+        app.config[str('MEMORY_CACHE_CONFIGURATION')].get(str('FETCH_S3_DATA_TIMEOUT'), 2 * 60)
+    )
+    def fetch_and_get_data(self, instance_name, bucket_name, folder, **kwargs):
+        result = dict()
         logger = logging.getLogger(__name__)
-        bucket_params = app.config.get('OLYMPIC_SITE_PARAMS_BUCKET', {})
-        if not bucket_params:
-            logger.debug("Reading stop points attractivities, undefined bucket_params")
-            return
-        bucket_name = bucket_params.get("name")
-        if not bucket_name:
-            logger.debug("Reading stop points attractivities, undefined bucket_name")
-            return
-
-        args = bucket_params.get(
-            "args", {"connect_timeout": 2, "read_timeout": 2, "retries": {'max_attempts': 0}}
-        )
-        s3_resource = boto3.resource('s3', config=Config(**args))
-
-        folder = app.config.get('OLYMPIC_SITE_PARAMS_BUCKET', {}).get("folder", "olympic_site_params")
+        s3_resource = boto3.resource('s3', config=Config(**kwargs))
         try:
             my_bucket = s3_resource.Bucket(bucket_name)
-            for obj in my_bucket.objects.filter(Prefix="{}/{}/".format(folder, self.instance.name)):
+            for obj in my_bucket.objects.filter(Prefix="{}/{}/".format(folder, instance_name)):
                 if obj.key.endswith('.json'):
-                    json_content = self.get_json_content(obj)
-                    self.str_datetime_time_stamp(json_content)
-                    self.olympic_site_params.update(json_content)
+                    resource_s3_object = ResourceS3Object(obj, instance_name)
+                    json_content = self.load_data(resource_s3_object)
+                    result.update(json_content)
         except Exception:
             logger.exception("Error on OlympicSiteParamsManager")
+        return result
+
+    def get_opg_params(self):
```

Comment:
minor: to be more Pythonic, you can define this as a `property`

Reply:
Done